### PR TITLE
API Updates

### PR DESCRIPTION
**Pull request by richardm-stripe**
Codegen for openapi 0f6948c.
r? @pakrym-stripe 
cc @stripe/api-libraries

## Changelog
Major version release. The [migration guide](https://github.com/stripe/stripe-php/wiki/Migration-Guide-for-v8) contains a detailed list of backwards-incompatible changes with upgrade instructions.

(⚠️ = breaking changes):
* ⚠️ Replace the legacy `Order` API with the new `Order` API. 
  * Resource modified: `Order`.
  * New methods: `cancel`, `list_line_items`, `reopen`, and `submit`
  * Removed methods: `pay` and `return_order`
  * Removed resources: `OrderItem` and `OrderReturn`
  * Removed references from other resources: `Charge.order`

* Add support for `amount_discount`, `amount_tax`, and `product` on `LineItem`
* Change type of `Charge.shipping.name`, `Checkout.Session.shipping.name`, `Customer.shipping.name`, `Invoice.customer_shipping.name`, `PaymentIntent.shipping.name`, `ShippingDetails.name`, and `Source.source_order.shipping.name` from `nullable(string)` to `string`

### CODE REVIEW

**Comment by pakrym-stripe** [comment]
I wonder if we need to add a link to a larger document describing Order changes to the changelog.

File: types/2020-08-27/Checkout/Sessions.d.ts [hunk 1]
```diff
@@ -588,7 +588,7 @@ declare module 'stripe' {
           /**
            * Recipient name.
            */
-          name?: string | null;
```
[the remark this answers is not in the document]
Comment (richardm-stripe) [reply types/2020-08-27/Checkout/Sessions.d.ts:591]: No, it is an unrelated strengthening of this type in our API definition.

[the remark this answers is not in the document]
Comment (pakrym-stripe) [reply types/2020-08-27/Checkout/Sessions.d.ts:591]: Was this an override before? Is there a PR with it's removal?

**Review by pakrym-stripe — APPROVED** [review]

File: types/2020-08-27/Checkout/Sessions.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (richardm-stripe) [reply types/2020-08-27/Checkout/Sessions.d.ts:591]: see slack

**Pull request merged**